lib/core/interactor/video_reader.rb

```diff
@@ -0,0 +1,36 @@
+module Core
+  class VideoReader
+    attr_accessor :id
+
+    private :id=
```

Comment:
Use `attr_reader` and in your initializer make the assignment via `@id = id` instead.